### DISCUSSION TITLE
Unexpected LED actions

### DISCUSSION THREAD

**Discussion by netdudeuk** [PlummersSoftwareLLC/NightDriverStrip#109] [General]
Firstly, thanks for sharing the software.

I downloaded the source code onto my MBP and I was able to build the _demo_ software for my ESP32 board.

There are only eight LEDs but the colours move along them nicely.

I also downloaded the source code onto my PC but I don't get the same result even though the relevant settings in the globals.h and platformio.ini files are the same.  I just get different LEDs turning on and off.

Any idea why please ?

Thanks.

**Comment by DanMan32** [comment]
That's what the demo does across 144 LEDs.  With just 8 LEDs, you won't get the same effect as you're only viewing the effects of the first 8 LEDs out of 144.

**Reply by netdudeuk** [reply]
It works ok when I download the same demo to the same ESP32 and the same LED strip from the Mac though.

These are some of the LED setup lines on my PC -

    #define MATRIX_WIDTH            8
    #define MATRIX_HEIGHT           1
    #define NUM_LEDS                (MATRIX_WIDTH*MATRIX_HEIGHT)
    #define NUM_CHANNELS            1
    #define NUM_RINGS               5
    #define RING_SIZE_0             24

This is what it looks like when programmed from thePC -

https://youtu.be/8JdZ_dpdznU

And by the Mac -

https://youtu.be/bsL3hB0D_jk

Here's the terminal debug output when programmed from the PC -

(I) (setup)(C1) Startup!
(I) (setup)(C1) Starting DebugLoopTaskEntry
(I) (DebugLoopTaskEntry)(C1) >> DebugLoopTaskEntry
(I) 
(I) (setup)(C1) Could allocate 3315 buffers but limiting it to 99
(I) 
(I) (setup)(C1) Reserving 99 LED buffers for a total of 56 bytes...
(I) (setup)(C1) Adding LEDs to FastLED...
(I) (setup)(C1) Adding 8 LEDs to FastLED.
(I) (setup)(C1) Initializing effects manager...
(I) (Init)(C1) First Effect: RGB Twinkle Stars
(I) (setup)(C1) Launching Drawing:
(E) (setup)(C1) Heap before launch: PASS
I) (NetworkHandlingLoopEntry)(C1) >Sampler Memory: 27(I)
(C1) Setup complete - ESP32 Free Memory: 271552
[0m(I)
I) (DrawLoopTaskEntry)(C0) Entry Heap: PASS
 271552
) (DrawLoopTaskEntry)(C0) Entry Heap: PASS
E) (DrawLE) (DrawLoopTaskEntry)(C0) Entry Heap: PASS
 271552
(E)
(I) (AudioSamplerTaskEntry)(C1) Mem: 271028 LED FPS: 142, Audio FPS: 40, gPeakVU:      0.00, gMinVU:      0.00, gVURatio: 0.0000, Peaks:     0.00,     0.00,     0.00,     0.00,     0.00
(I) (AudioSamplerTaskEntry)(C1) Mem: 270956 LED FPS: 142, Audio FPS: 40, gPeakVU:      0.00, gMinVU:      0.00, gVURatio: 0.0000, Peaks:     0.00,     0.00,     0.00,     0.00,     0.00
(I) (AudioSamplerTaskEntry)(C1) Mem: 270436 LED FPS: 142, Audio FPS: 40, gPeakVU:      0.00, gMinVU:      0.00, gVURatio: 0.0000, Peaks:     0.00,     0.00,     0.00,     0.00,     0.00
(I) (AudioSamplerTaskEntry)(C1) Mem: 270952 LED FPS: 142, Audio FPS: 40, gPeakVU:      0.00, gMinVU:      0.00, gVURatio: 0.0000, Peaks:     0.00,     0.00,     0.00,     0.00,     0.00


And different for the Mac version -

(I) (setup)(C1) Startup!
(I) (setup)(C1) Starting DebugLoopTaskEntry
(I) (DebugLoopTaskEntry)(C1) >> DebugLoopTaskEntry
(I) 
(I) (setup)(C1) Could allocate 3451 buffers but limiting it to 99
(I) 
(I) (setup)(C1) Reserving 99 LED buffers for a total of 56 bytes...
(I) (setup)(C1) Adding LEDs to FastLED...
(I) (setup)(C1) Adding 8 LEDs to FastLED.
(I) (setup)(C1) Initializing effects manager...
(I) (Init)(C1) First Effect: Palette Effect
(I) (setup)(C1) Launching Drawing:
(E) (setup)(C1) Heap before launch: PASS
(I) (DrtworkHandlingLoopEntry)(C1) >> NetworkHandlingL(I) (NetworkHandlingLoopEntry)(C1) >> NetworkHandlingLoopEntry
(I) 
(I) (setup(I) (DrawLoopTaskEntry)(C1) Entry Heap: PASS
 289928
(E) (Dra0mopTaskEntry)(C1) Entry Heap: PASS
(E) 
(I) (CheckEffectTimerExpired)(C1) 30001ms elapsed: Next Effect
(I) (CheckEffectTimerExpired)(C1) Current Effect: Palette Effect
(I) (CheckEffectTimerExpired)(C1) 30000ms elapsed: Next Effect
(I) (CheckEffectTimerExpired)(C1) Current Effect: Palette Effect
